### PR TITLE
Part 5: Wrap controllers in scheduling/tenancy pkgs

### PR DESCRIPTION
**Pull request by varshaprasad96**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/server/controllers.go [hunk 1]
```diff
@@ -849,18 +850,13 @@ func (s *Server) installWorkloadNamespaceScheduler(ctx context.Context, config *
 
 func (s *Server) installSchedulingPlacementController(ctx context.Context, config *rest.Config, server *genericapiserver.GenericAPIServer) error {
 	controllerName := "kcp-scheduling-placement-controller"
-	config = rest.AddUserAgent(rest.CopyConfig(config), controllerName)
-	kubeClusterClient, err := kubernetes.NewClusterForConfig(config)
```
Comment (varshaprasad96) [inline pkg/server/controllers.go:853]: `kubeClusterClient` didn't seem to be used anywhere in this controller.

File: pkg/reconciler/tenancy/clusterworkspacedeletion/clusterworkspace_deletion_controller.go [hunk 2]
```diff
@@ -45,8 +45,8 @@ import (
 )
 
 func NewController(
-	kcpClient kcpclient.ClusterInterface,
-	metadataClusterClient metadata.ClusterInterface,
+	kcpClient kcpclient.Interface,
```
Comment (sttts) [inline pkg/reconciler/tenancy/clusterworkspacedeletion/clusterworkspace_deletion_controller.go:48]: clusterClient

File: pkg/reconciler/tenancy/clusterworkspacedeletion/clusterworkspace_deletion_controller.go [hunk 3]
```diff
@@ -81,8 +81,8 @@ func NewController(
 type Controller struct {
 	queue workqueue.RateLimitingInterface
 
-	kcpClient             kcpclient.ClusterInterface
-	metadataClusterClient metadata.ClusterInterface
+	kcpClient             kcpclient.Interface
```
Comment (sttts) [inline pkg/reconciler/tenancy/clusterworkspacedeletion/clusterworkspace_deletion_controller.go:84]: clusterClient

**Pull request merged**